frontend/src/store/modules/sheet.ts

```diff
@@ -146,16 +149,20 @@ export const useSheetStore = defineStore("sheet", {
       }
       this.sheetById.set(sheetId, sheet);
     },
-    deleteSheetById(sheetId: SheetId) {
-      const idx = this.sheetList.findIndex((sheet) => sheet.id === sheetId);
-      if (idx !== -1) this.sheetList.splice(idx, 1);
+    upsertSheet(payload: {
+      sheet: Pick<Sheet, "id" | "name" | "statement">;
+      currentTab: TabInfo;
+    }): Promise<Sheet> {
+      const { sheet, currentTab } = payload;
+      const hasSheet = this.sheetById.has(sheet.id);
 
-      if (this.sheetById.has(sheetId)) {
-        this.sheetById.delete(sheetId);
+      if (hasSheet) {
+        return this.patchSheetById(sheet);
+      } else {
```

Comment:
"else" is unnecessary under this context.

Reply:
Yes, we commonly use "short circuit return" to avoid the second (unnecessary) branch.